controller/api/public/stat_summary.go

```diff
@@ -329,6 +459,37 @@ func buildRequestLabels(req *pb.StatSummaryRequest) (labels model.LabelSet, labe
 	return
 }
 
+func buildTrafficSplitRequestLabels(req *pb.StatSummaryRequest) (labels model.LabelSet, labelNames model.LabelNames) {
+	// labelNames: the group by in the prometheus query
+	// labels: the labels for the resource we want to query for
+
+	switch out := req.Outbound.(type) {
+	case *pb.StatSummaryRequest_ToResource:
+		labelNames = promGroupByLabelNames(req.Selector.Resource)
+
+		labels = labels.Merge(promDstQueryLabels(out.ToResource))
+		labels = labels.Merge(promQueryLabels(req.Selector.Resource))
+		labels = labels.Merge(promDirectionLabels("outbound"))
+
+	case *pb.StatSummaryRequest_FromResource:
+		labelNames = promDstGroupByLabelNames(req.Selector.Resource)
+
+		labels = labels.Merge(promQueryLabels(out.FromResource))
+		labels = labels.Merge(promDstQueryLabels(req.Selector.Resource))
+		labels = labels.Merge(promDirectionLabels("outbound"))
+
+	default:
+		labelNames = promGroupByLabelNames(req.Selector.Resource)
+
+		labels = labels.Merge(promQueryLabels(req.Selector.Resource))
+		labels = labels.Merge(promDirectionLabels("outbound"))
+	}
+
+	labelNames[1] = model.LabelName("dst_service") // replacing "trafficsplit" with "dst_service"
```

Comment:
I'm a bit confused on what this is doing.

Reply:
I refactored this, let me know if it's more clear. `buildTrafficSplitRequestLabels` handles four different query scenarios, and needs to output different labels for each: 

`stat ts -A` (namespace is not specified)
`stat ts -n foo`
`stat ts --from resource/name`
`stat ts --to resource/name`